extensions/amp-sidebar/1.0/component.js

```diff
@@ -190,45 +191,46 @@ export function SidebarToolbar({
   children,
   ...rest
 }) {
-  const ref = useRef();
-  const [matches, setMatches] = useState(false);
-  const [targetEl, setTargetEl] = useState(null);
+  const ref = useRef(null);
+  const [target, setTarget] = useState();
 
   useEffect(() => {
-    const window = ref.current?.ownerDocument?.defaultView;
-    if (!window) {
-      return;
-    }
-
-    const mediaQueryList = window.matchMedia(mediaQueryProp);
-    const updateMatches = () => setMatches(mediaQueryList.matches);
-    mediaQueryList.addEventListener('change', updateMatches);
-    setMatches(mediaQueryList.matches);
-    return () => mediaQueryList.removeEventListener('change', updateMatches);
-  }, [mediaQueryProp]);
+    const selector = `#${escapeCssSelectorIdent(toolbarTarget)}`;
+    const newTarget = document.querySelector(selector);
+    setTarget(newTarget);
+  }, [toolbarTarget]);
 
   useEffect(() => {
-    const document = ref.current?.ownerDocument;
-    if (!document) {
+    const element = ref.current;
+    const doc = ref.current?.ownerDocument;
+    if (!element || !doc || !target) {
```

Comment:
We should not need to check both `!element` and `!doc` since `doc` will not be defined if `element` isn't.
```suggestion
    if (!doc || !target) {
```

Reply:
Thanks Caroline, great suggestion as always!